samples/snippets/src/main/java/com/example/spanner/PgSpannerSample.java

```diff
@@ -0,0 +1,702 @@
+      // If the operation failed during execution, expose the cause.
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      // Throw when a thread is waiting, sleeping, or otherwise occupied,
+      // and the thread is interrupted, either before or during the activity.
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_create_storing_index]
+
+  // Before running this example, create a storing index AlbumsByAlbumTitle2 by applying the DDL
+  // statement "CREATE INDEX AlbumsByAlbumTitle2 ON Albums(AlbumTitle) STORING (MarketingBudget)".
+  // [START spanner_read_data_with_storing_index]
+  static void readStoringIndex(DatabaseClient dbClient) {
+    // We can read MarketingBudget also from the index since it stores a copy of MarketingBudget.
+    try (ResultSet resultSet =
+             dbClient
+                 .singleUse()
+                 .readUsingIndex(
+                     "Albums",
+                     "AlbumsByAlbumTitle2",
+                     KeySet.all(),
+                     Arrays.asList("AlbumId", "AlbumTitle", "MarketingBudget"))) {
+      while (resultSet.next()) {
+        System.out.printf(
+            "%d %s %s\n",
+            resultSet.getLong(0),
+            resultSet.getString(1),
+            // TODO: MarketingBudget uppercase not supported, keeping lowercase for now
+            resultSet.isNull("marketingbudget") ? "NULL" : resultSet.getLong(2));
+      }
+    }
+  }
+  // [END spanner_read_data_with_storing_index]
+
+  // [START spanner_read_only_transaction]
+  static void readOnlyTransaction(DatabaseClient dbClient) {
+    // ReadOnlyTransaction must be closed by calling close() on it to release resources held by it.
+    // We use a try-with-resource block to automatically do so.
+    try (ReadOnlyTransaction transaction = dbClient.readOnlyTransaction()) {
+      ResultSet queryResultSet =
+          transaction.executeQuery(
+              Statement.of("SELECT SingerId, AlbumId, AlbumTitle FROM Albums"));
+      while (queryResultSet.next()) {
+        System.out.printf(
+            "%d %d %s\n",
+            queryResultSet.getLong(0), queryResultSet.getLong(1),
+            queryResultSet.getString(2));
+      }
+      try (ResultSet readResultSet =
+               transaction.read(
+                   "Albums", KeySet.all(), Arrays.asList("SingerId", "AlbumId", "AlbumTitle"))) {
+        while (readResultSet.next()) {
+          System.out.printf(
+              "%d %d %s\n",
+              readResultSet.getLong(0), readResultSet.getLong(1),
+              readResultSet.getString(2));
+        }
+      }
+    }
+  }
+  // [END spanner_read_only_transaction]
+
+  // [START spanner_query_singers_table]
+  static void querySingersTable(DatabaseClient dbClient) {
+    try (ResultSet resultSet =
+             dbClient
+                 .singleUse()
+                 .executeQuery(Statement.of("SELECT singerid as \"SingerId\", "
```

Comment:
@ansh0l , I do not know much how the client handle the identifier, if we just use SELECT SingerId ...

would the resultSet.getLong("SingerId")  work or not?

Reply:
@jin-jj No, it would not work. If you execute `SELECT SingerId FROM Singers`, then PostgreSQL will fold the identifier to lower case, so it is effectively the same as executing `SELECT singerid FROM Singers`.

Identifiers that are quoted will be treated as case-sensitive and keep the actual lower/upper case letters.